main.ts

```diff
@@ -109,27 +107,22 @@ export default class FilenameHeadingSyncPlugin extends Plugin {
       return;
     }
 
-    const editor = view.editor;
-    const doc = editor.getDoc();
-    const docstart = this.findDocstart(doc);
-    const heading = this.findHeading(doc, docstart);
+    this.app.vault.read(file).then((data) => {
+      const lines = data.split('\n');
+      const start = this.findNoteStart(lines);
+      const heading = this.findHeading(lines, start);
 
-    // no heading found, nothing to do here
-    if (heading == null) {
-      return;
-    }
+      if (heading === null) return; // no heading found, nothing to do here
 
-    const sanitizedHeading = this.sanitizeHeading(heading.Text);
-    if (
-      sanitizedHeading.length > 0 &&
-      this.sanitizeHeading(view.file.basename) !== sanitizedHeading
-    ) {
-      const newPath = view.file.path.replace(
-        view.file.name.trim(),
-        `${sanitizedHeading}.${view.file.extension}`,
-      );
-      this.app.fileManager.renameFile(view.file, newPath);
-    }
+      const sanitizedHeading = this.sanitizeHeading(heading.text);
+      if (
+        sanitizedHeading.length > 0 &&
+        this.sanitizeHeading(file.basename) !== sanitizedHeading
+      ) {
+        const newPath = file.path.replace(file.basename, sanitizedHeading);
+        this.app.vault.rename(file, newPath);
```

Comment:
I used `renameFile` here because `rename` did not update links in other files. Is that still the case?

Reply:
Ah, I very much missed this functionality, that is indeed the case! Will switch this back to the FileManager's renameFile method.